### PR TITLE
MultiRewards distributor: rewarder allowlist

### PR DESCRIPTION
**Pull request by gtaschuk**
Uses a mapping rather than an `EnumerableSet` to more efficiently manage list of rewarders allowed to distribute in the staking contract

Renames whitelist -> allowlist re: @FernandoMartinelli 's point about outdated language https://balancerlabs.slack.com/archives/C0146NRBJEP/p1624636747191400

### CODE REVIEW

File: pkg/asset-manager-utils/contracts/AaveATokenAssetManager.sol [hunk 1]
```diff
@@ -60,7 +60,7 @@ contract AaveATokenAssetManager is RewardsAssetManager {
 
         distributor = IMultiRewards(rewardsDistributor);
         IERC20 poolAddress = IERC20(uint256(poolId) >> (12 * 8));
-        distributor.whitelistRewarder(poolAddress, stkAave, address(this));
+        distributor.allowlistRewarder(poolAddress, stkAave, address(this));
```
Comment (nventuro) [inline pkg/asset-manager-utils/contracts/AaveATokenAssetManager.sol:63]: Doesn't 'allow' make more  sense as a verb? Also e.g. 'only allowed rewarder', 'is allowed rewarder', etc.

Comment (gtaschuk) [reply pkg/asset-manager-utils/contracts/AaveATokenAssetManager.sol:63]: :thinking: 
_tl;dr - I agree the term feels unfamiliar but I would suggest we use allowlist as a 1:1 standin for whitelist_

**Whitelist** is commonly understood as both a verb or a noun, which is why `whitelistRewarder` feels familiar.  **Allowlist** as a word is just not common in spoken english (yet?) but is being used as a direct stand-in in code.  I agree that the verb usage, especially, feels a little cumbersome, but I think that's just because it's uncommon today.

It is however becoming much more common, especially among larger companies.  Google, Microsoft, Youtube, VMWare etc are all replacing "whitelist" with "allowlist" so I suspect it will feel more familiar over time.

ie here's a vmware glossary definition about the practice of allowlisting (verb usage)
https://www.vmware.com/topics/glossary/content/allowlisting

I don't really like `allowRewarder` because it feels vague ("allow Rewarders to do what?").  The purpose of the function is to add the rewarder to the allowlist, and `allowlistRewarder` feels better than `addAddressToRewarderAllowlist`

Comment (nventuro) [reply pkg/asset-manager-utils/contracts/AaveATokenAssetManager.sol:63]: I was going to say that it might feel too jarring, but in the end I think you're right. 'allow' doesn't convey the same meaning of 'add to the set of allowed actors'. Thanks for the writeup!

**Review by nventuro — APPROVED** [review]
